src/cli.rs

```diff
@@ -1276,6 +1377,58 @@ impl<'a> BirliContext<'a> {
         Ok(result)
     }
 
+    /// Call `run()` for every channel range in `self.channel_range_sel`.
+    pub fn run_ranges(self) -> Vec<Result<HashMap<String, Duration>, BirliError>> {
+        let ranges = self.channel_range_sel.ranges.clone();
```

Comment:
This clone can probably be avoided by iterating over `&self.channel_range_sel.ranges` on line 1400.